src/main/java/org/scijava/nativelib/BaseJniExtractor.java

```diff
@@ -271,25 +263,25 @@ File extractResource(final File dir, final URL resource,
 	 * Another issue is that createTempFile only guarantees to use the first three
 	 * characters of the prefix, so I could delete a similarly-named temporary
 	 * shared library if I haven't loaded it yet.
-	 *
-	 * @param prefix
-	 * @param suffix
 	 */
-	void deleteLeftoverFiles(final String prefix, final String suffix) {
-		final File tmpDirectory = new File(System.getProperty(JAVA_TMPDIR));
+	void deleteLeftoverFiles() {
+		final File tmpDirectory = new File(System.getProperty(JAVA_TMPDIR, ALTR_TMPDIR));
 		final File[] files = tmpDirectory.listFiles(new FilenameFilter() {
 
 			public boolean accept(final File dir, final String name) {
-				return name.startsWith(prefix) && name.endsWith(suffix);
+				return name.startsWith(TMP_PREFIX);
 			}
 		});
 		if (files == null) return;
 		for (final File file : files) {
 			// attempt to delete
 			try {
+				Files.walk(file.toPath()).map(Path::toFile)
+						.sorted(Comparator.reverseOrder())
+						.forEachOrdered(File::delete);
```

Comment:
The purpose of this is to delete recursively? If so, you can [steal this code](https://github.com/scijava/scijava-common/blob/scijava-common-2.69.0/src/main/java/org/scijava/util/FileUtils.java#L489-L508) to do it, without depending on Java 8.